### ISSUE TITLE
Enhancement: azurerm_databricks_workspace Private Network

### ISSUE THREAD

**Issue by a138076** [terraform-providers/terraform-provider-azurerm#3001]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Enhance the existing Azure Databricks resource to allow configuration of the resource within a Private Virtual Network

### New or Affected Resource(s)

* azurerm_databricks_workspace

### Potential Terraform Configuration

```hcl
resource "azurerm_databricks_workspace" "test" {
  name                 = "databricks-test"
  resource_group_name  = "${azurerm_resource_group.test.name}"
  location             = "${azurerm_resource_group.test.location}"
  sku                  = "standard"
  virtual_network_id   = "<networkid>"
  public_subnet_name   = "public"
  public_subnet_range  = "10.0.0.0/28"
  private_subnet_name  = "private"
  private_subnet_range = "10.0.1.0/28"
}
```

### References

* https://github.com/Azure/azure-quickstart-templates/blob/master/101-databricks-workspace-with-vnet-injection/azuredeploy.json
* https://docs.azuredatabricks.net/administration-guide/cloud-configurations/azure/vnet-inject.html

**Comment by sijusamueltech** [comment]
which azure rm version will have this feature

**Comment by a138076** [comment]
@sijusamueltech It's currently in preview at the moment already

**Comment by sijusamueltech** [comment]
> @sijusamueltech It's currently in preview at the moment already

Thanks for the info.  How do we get the preview version of the azurerm ?  that would be really helpful  ( Do we need to get a specific  branch and build it manually).   Thx

**Comment by sijusamueltech** [comment]
Error: azurerm_databricks_workspace.databricks: : invalid or unknown key: public_subnet_range
Terraform v0.11.13
+ provider.azurerm v1.23.0

**Comment by tombuildsstuff** [comment]
@sijusamueltech unfortunately this functionality isn't currently natively supported by Terraform (which is why this issue is still open) - however updates will be posted to this issue when support for this becomes available :)

**Comment by arsenyspb** [comment]
@tombuildsstuff when planning this issue... can you please add the ability to point to a link of an ARM template, instead of **only** taking in input related to VNet/Subnet etc. in the variables?.. User story is -- there are couple of other parameters that might be often times specified when provisioning Workspace... some users, due to data governance, prefer to have Locally-Redundant Storage for the Storage Account created inside managed resource group for the root of [DBFS](https://docs.azuredatabricks.net/user-guide/dbfs-databricks-file-system.html)... therefore, specifying ` "storageAccountSkuName": { "value": "Standard_LRS" } ` as workspace property... And maybe for future cases... Just i.e. be able to take in the same JSON reference that @a138076 mentioned ([here](https://github.com/Azure/azure-quickstart-templates/blob/master/101-databricks-workspace-with-vnet-injection/azuredeploy.json))

**Comment by tombuildsstuff** [comment]
@arsenyspb unfortunately that approach isn't supported by the API (since we map the properties from the SDK/API) - as such we're unable to do so.

Rather than exposing an ARM Template (which could potentially fail during deployment, leaving spurious resources) - it'd be better for the API to expose a new optional field for the ID of an existing Storage Account which could be used rather than a new one being created; but AFAIK this doesn't exist at this time and would be a feature request to the API team.

**Comment by arsenyspb** [comment]
@tombuildsstuff understood... I might have misread the [(5) in Deploy with the REST API](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-deploy-rest#deploy-with-the-rest-api) -- I thought RP is able concatenate / nest the JSON in to the payload of request. 
So, if we focus on the properties, can we then add at least ability to specify the said`"storageAccountSkuName":` { "value": "Standard_LRS" }` (could be omitted by default)., -- what do you think?..

**Comment by arsenyspb** [comment]
And oi, no, @tombuildsstuff -- I was not talking about existing Storage Account, -- no, it's the property of a newly-created storage account as part of the new Workspace deployment (the root for DBFS inside Managed Group)

**Comment by tombuildsstuff** [comment]
@arsenyspb to clarify: Terraform doesn't deploy an ARM Template (ignoring the `azurerm_template_deployment` resource which does 🙃) - instead we use [the Azure SDK for Go](https://github.com/Azure/azure-sdk-for-go) (and it's structs) to interact with the API; so these properties would need to be exposed there for us to add support for this :)

**Comment by alagala** [comment]
@arsenyspb As a temporary workaround (until Terraform and Azure SDK for Go will support VNET injection and LRS storage account), I have setup a Terraform module that calls a Azure Databricks ARM template to enable both LRS and VNET injection.
If interested, check it out at https://github.com/alagala/labs/tree/master/azure/databricks

**Comment by schlbra** [comment]
Any updates on this issue or the most recent PR? @tombuildsstuff @notchairmk

**Comment by tombuildsstuff** [comment]
@schlbra whilst this is fixed in the Swagger this is now waiting on the SDK upgrade, which'll be available via #5335 - once that's done it should be possible for someone to take a look into this :)

**Comment by michaelmcmillan** [comment]
Is this now available to be used?

**Comment by schlbra** [comment]
@michaelmcmillan , it has been included in version 1.42.0 which has not been released yet.
